lib/teach-api.js

```diff
@@ -73,6 +73,20 @@ _.extend(TeachAPI.prototype, {
     }
 
     return req;
+  },
+  getAllData: function(callback) {
+    request('get', process.env.TEACH_API_URL || this.baseURL + "/api/clubs")
+      .accept('json')
+      .end(function(err, res) {
+        if(err) {
+          return callback(err);
+        }
+        try {
+          callback(null, JSON.parse(res.text));
+        } catch(e) {
+          return callback(e);
```

Comment:
No need for this try/catch since `res.body` will already be the parsed content, so there's no `JSON.parse` to throw an exception!